src/core/src/macros/mod.rs

```diff
@@ -30,14 +30,14 @@ macro_rules! gfx_vertex {
                 use $crate::attrib::{Offset, Stride};
                 use $crate::attrib::format::ToFormat;
                 let stride = size_of::<$name>() as Stride;
-                let mut offset = 0 as Offset;
+                let tmp: $name = unsafe{ ::std::mem::uninitialized() };
```

Comment:
My concern with this is that if `$name` supports `drop`. It's possible for this to lead to undefined behavior. But at the same time, I can't imagine anyone using `gfx_vertex!` for anything other then Plain-Old-Data.

It could possibly be made safer by adding `std::mem::forget(tmp)` at the end. That's actually not quite enough, because a `panic!()` caused by `<$ty as ToFormat>::describe()` in the case where a user is doing something _very_ strange could happen.

If we move the creation around tmp just before line 37. and do a forget on line 43. I think it is safe, even from destructors.

But I might just be paranoid.